submissions/israelOladosu/DomManipulation/Script.js

```diff
@@ -0,0 +1,107 @@
+let submitForm = document.getElementById("validateForm"),
+  email = document.getElementById("email"),
+  password = document.getElementById("password"),
+  confirmPassword = document.getElementById("pswd"),
+  firstName = document.getElementById("firstName"),
+  lastName = document.getElementById("lastName"),
+  otherName = document.getElementById("otherName"),
+  phoneNumber = document.getElementById("phone");
+
+var regEx = {
+  name: /[^A-Za-z]/,
+  password: /^(?=.*[A-Z])(?=.*[a-z])(?=.*[@$!%*?&_<>':;])(?=.*[\d])[a-zA-Z0-9@$!%*?&_<>':;]{6,}$/,
+  email: /^([a-z\d\.-]+)@([a-z\d-]+)\.([a-z]{2,8})$/i,
+  phoneNumber: /^[\+0-9]{11,}$/,
+};
+
+function required(field, e) {
+  if (field.id == "pswd" && field.value == "") {
+    field.nextElementSibling.innerHTML = `Please confirm your password`;
+  } else if (field.value == "") {
+    field.nextElementSibling.innerHTML = `Please enter your ${field.previousElementSibling.innerText}`;
+    e.preventDefault();
+    return false;
+  } else {
+    field.nextElementSibling.innerHTML = "";
+    return true;
+  }
+}
+
+function passwordValidation(pswd1, pswd2, e) {
+  if (pswd1.value !== "") {
+    if (pswd1.id === "password" && !regEx.password.test(pswd1.value)) {
+      pswd1.nextElementSibling.innerHTML =
+        "Your password should have a minimum of 5 characters, 1 capital letter, 1 special character eg @ and 1 number.";
+      e.preventDefault();
+      return false;
+    } else if (pswd1.value.trim() !== pswd2.value.trim()) {
+      pswd1.nextElementSibling.innerHTML = "Your passwords don't match";
+      pswd2.nextElementSibling.innerHTML = "Your passwords don't match";
+      e.preventDefault();
+      return false;
+    } else {
+      pswd1.nextElementSibling.innerHTML = "";
+      pswd2.nextElementSibling.innerHTML = "";
+      return true;
+    }
+  }
+}
+
+function nameValidation(name, e) {
+  if (name.value !== "") {
+    if (
+      (name.id === "firstName" || name.id === "lastName") &&
+      name.value.match(regEx.name)
+    ) {
+      name.nextElementSibling.innerHTML = `${name.previousElementSibling.innerText} must not contain special character`;
+      e.preventDefault();
+      return false;
+    } else {
+      email.nextElementSibling.innerHTML = "";
+      return true;
+    }
+  }
+}
+
+function emailValidation(email, e) {
+  if (email.value !== "") {
+    if (!regEx.email.test(email.value)) {
+      email.nextElementSibling.innerHTML = "Please enter a valid email";
+      e.preventDefault();
+      return false;
+    } else {
+      email.nextElementSibling.innerHTML = "";
+      return true;
+    }
+  }
+}
+
+function phoneNumberValidation(digit, e) {
+  if (digit.value !== "") {
```

Comment:
You didn't fix this

Reply:
I wish you could run the code with the form fields.
If i don't use that logic, no output will be shown if a user tries to submit an empty field and this is because the other validatons where called after required and since its the same field we are passing to the validation functions it clears out whatever required is returning and also not that the validation function are not checking for empty field. So what that logic does is to ensure that the validation functions are fired only when there is an input else if it fires after required based on the arrangements in validateForm function. I hope you get my long explaination.

They ain't useless.
I wish I coukd call you to explain better

Reply:
I sure get your point. That why you need to call the required function first so that one will be executed first before that.